### PR TITLE
fix(angular): share secondary entry points

### LINKED ISSUE #9863: NullInjectorError: No provider for InjectionToken DocumentToken! when using BrowserAnimationsModule in an Angular MFE workspace
## Current Behavior
When importing either `BrowserAnimationsModule` or `NoopAnimationsModule`, the app fails with the error: 
`NullInjectorError: R3InjectorError(AppModule)[RendererFactory2 -> AnimationEngine -> InjectionToken DocumentToken -> InjectionToken DocumentToken -> InjectionToken DocumentToken]: 
  NullInjectorError: No provider for InjectionToken DocumentToken!`

## Expected Behavior
Expect the workspace to share dependancies between apps and libraries. Expect to able to use Angular Animations without breaking the app. 

### Is this a regression? .i.e Did this used to be the behavior at one point?
Yes. Before the update to the webpack config file in 13.10.2, we could share internal dependancies with the sharedMappings plugin, and external dependancies with mf.share(). That seems to no longer be the case (or I just don't know how to).

## Steps to Reproduce
* Create an empty nx workspace (`npx create-nx-workspace`, choose the `apps` option for an empty workspace)
* Add the @nrwl/angular plugin (`npm install --save-dev @nrwl/angular`)
* Create a shell app (`npx nx g @nrwl/angular:host container`)
* Create a child app (`npx nx g @nrwl/angular:remote childApp --host=container`)
* Install @angular/animations (`npm install --save @angular/animations`)
* Import BrowserAnimationsModule or NoopAnimationsModule in the AppModule of the Shell app
* Run the apps (nx run container:serve-mfe)

Github repo reproducing the issue: https://github.com/Mike-Mountain/nx-issue

### Failure Logs

NullInjectorError: R3InjectorError(AppModule)[RendererFactory2 -> AnimationEngine -> InjectionToken DocumentToken -> InjectionToken DocumentToken -> InjectionToken DocumentToken]: 
NullInjectorError: No provider for InjectionToken DocumentToken!

### Environment   
```
Node : 16.13.2
OS   : darwin x64
npm  : 8.1.2
nx : 13.10.2
@nrwl/angular : 13.10.2
@nrwl/cypress : 13.10.2
@nrwl/detox : Not Found
@nrwl/devkit : 13.10.2
@nrwl/eslint-plugin-nx : 13.10.2
@nrwl/express : Not Found
@nrwl/jest : 13.10.2
@nrwl/js : Not Found
@nrwl/linter : 13.10.2
@nrwl/nest : Not Found
@nrwl/next : Not Found
@nrwl/node : Not Found
@nrwl/nx-cloud : Not Found
@nrwl/nx-plugin : Not Found
@nrwl/react : Not Found
@nrwl/react-native : Not Found
@nrwl/schematics : Not Found
@nrwl/storybook : 13.10.2
@nrwl/web : Not Found
@nrwl/workspace : 13.10.2
typescript : 4.6.3
rxjs : 7.4.0
---------------------------------------
Community plugins:
```

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior
Secondary entry points of packages are not being shared correctly

## Expected Behavior

Secondary entry points of packages should be shared

Fixes #9863

### CODE REVIEW

File: packages/angular/src/utils/mfe/mfe-webpack.ts [hunk 1]
```diff
@@ -81,6 +81,21 @@ export function shareWorkspaceLibraries(
   };
 }
 
+function collectPackageSecondaries(pkgName: string, packages: string[]) {
+  const pathToPackage = join(workspaceRoot, 'node_modules', pkgName);
+  const secondaries = readdirSync(pathToPackage)
+    .map((file) => join(pathToPackage, file))
+    .filter((file) => lstatSync(file).isDirectory() && file !== 'node_modules')
```
Comment (leosvelperez) [inline packages/angular/src/utils/mfe/mfe-webpack.ts:88]: Given we are mapping to `join(pathToPackage, file)`, unless `pathToPackage` is an empty string (which it shouldn't), it doesn't seem `file` here could ever match `node_modules`, isn't it? Should this be using `endsWith` or filtering out `node_modules` before the mapping occurs?

File: packages/angular/src/utils/mfe/mfe-webpack.ts [hunk 2]
```diff
@@ -81,6 +81,21 @@ export function shareWorkspaceLibraries(
...
+    .filter((dir) => existsSync(join(dir, 'package.json')));
+  for (const secondary of secondaries) {
+    const libName = secondary
+      .replace(/\\/g, '/')
+      .replace(/^.*node_modules[/]/, '');
```
Comment (leosvelperez) [inline packages/angular/src/utils/mfe/mfe-webpack.ts:93]: Might be safer to rely on the `name` specified in the secondary entry point's `package.json`.

Comment (Coly010) [reply packages/angular/src/utils/mfe/mfe-webpack.ts:93]: that's a great point

File: packages/angular/src/utils/mfe/mfe-webpack.ts [hunk 3]
```diff
@@ -81,6 +81,21 @@ export function shareWorkspaceLibraries(
   };
 }
 
+function collectPackageSecondaries(pkgName: string, packages: string[]) {
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/utils/mfe/mfe-webpack.ts:84]: I think that might be an edge case. is that even supported in npm?

File: packages/angular/src/utils/mfe/mfe-webpack.ts (hunk 1 above)
Comment (Coly010) [reply packages/angular/src/utils/mfe/mfe-webpack.ts:88]: yes you're right

File: packages/angular/src/utils/mfe/mfe-webpack.ts [hunk 4]
```diff
@@ -81,6 +81,21 @@ export function shareWorkspaceLibraries(
...
+  const pathToPackage = join(workspaceRoot, 'node_modules', pkgName);
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/utils/mfe/mfe-webpack.ts:85]: ah ok, i've never used that before, so hopefully that will help

[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/utils/mfe/mfe-webpack.ts:85]: Can't use this it seem, it's pointing to built files: `/@angular/common/fesm2015/common.mjs`

**Pull request merged**